src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -623,6 +632,10 @@ bool StatefulReader::processDataFragMsg(
             {
                 mp_history->completed_change(work_change);
                 pWP->received_change_set(work_change->sequenceNumber);
+                if (data_filter_ && !data_filter_->is_relevant(*work_change, m_guid))
+                {
+                    mp_history->remove_change(work_change);
```

Comment:
Shouldn't we return without notifying?

Reply:
No, because we could have filtered a change that is just before another one that should be notified.